rust/src/basis.rs

```diff
@@ -0,0 +1,211 @@
+use crate::prelude::*;
+
+/// CeedBasis context wrapper
+pub struct Basis<'a> {
+    ceed: &'a crate::Ceed,
+    ptr: bind_ceed::CeedBasis,
+}
+
+pub enum QuadMode {
+    Gauss,
+    GaussLobatto,
+}
+pub enum ElemTopology {
+    Line,
+    Triangle,
+    Quad,
+    Tet,
+    Pyramid,
+    Prism,
+    Hex,
+}
+pub enum EvalMode {
+    None,
+    Interp,
+    Grad,
+    Div,
+    Curl,
+    Weight,
+}
```

Comment:
Do we want to just put all enums in `ceed` like was done with the Python interface? Then the user always uses `ceed::EnumType::EnumValue` like `ceed::MemType::Host`